src/System.Diagnostics.DiagnosticSource/src/System/Diagnostics/DiagnosticSourceEventSource.cs

```diff
@@ -374,6 +480,29 @@ public FilterAndTransform(string filterAndPayloadSpec, int startIdx, int endIdx,
                     }
                 }
 
+                Action<string, string, IEnumerable<KeyValuePair<string, string>>> writeEvent = null;
+                if (activityName != null && activityName.Contains("Activity"))
+                {
+                    MethodInfo writeEventMethodInfo = typeof(DiagnosticSourceEventSource).GetTypeInfo().GetDeclaredMethod(activityName);
```

Comment:
This code only runs when you turn a particular DiagnosticSource On with event name renaming.   My expectation is that this happens rarely (as many times as a particular listener declares an activity that he cares about).  Thus I don't think caching is warranted at this time.   (Simplicity by default).